examples/arkode/CXX_manyvector/ark_sod_lsrk.cpp

```diff
@@ -0,0 +1,571 @@
+      if (check_flag(flag, "ARKodeSetStopTime")) { return 1; }
+    }
+
+    //   Advance in time
+    flag = ARKodeEvolve(arkode_mem, tout, y, &t, ARK_NORMAL);
+    if (check_flag(flag, "ARKodeEvolve")) { break; }
+
+    // Output solution
+    flag = WriteOutput(t, y, udata, uopts);
+    if (check_flag(flag, "WriteOutput")) { return 1; }
+
+    // Update output time
+    tout += dTout;
+    tout = (tout > udata.tf) ? udata.tf : tout;
+  }
+
+  // Close output
+  flag = CloseOutput(uopts);
+  if (check_flag(flag, "CloseOutput")) { return 1; }
+
+  // ------------
+  // Output stats
+  // ------------
+
+  if (uopts.output)
+  {
+    cout << "Final integrator statistics:" << endl;
+    flag = ARKodePrintAllStats(arkode_mem, stdout, SUN_OUTPUTFORMAT_TABLE);
+  }
+
+  // --------
+  // Clean up
+  // --------
+
+  ARKodeFree(&arkode_mem);
+  for (int i = 0; i < NSPECIES; i++) { N_VDestroy(vecs[i]); }
+  N_VDestroy(y);
+
+  return 0;
+}
+
+// -----------------------------------------------------------------------------
+// Functions called by the integrator
+// -----------------------------------------------------------------------------
+
+// ODE RHS function
+int frhs(sunrealtype t, N_Vector y, N_Vector f, void* user_data)
+{
+  // Access problem data
+  EulerData* udata = (EulerData*)user_data;
+
+  // initialize output to zeros
+  N_VConst(ZERO, f);
+
+  // Access data arrays
+  sunrealtype* rho = N_VGetSubvectorArrayPointer_ManyVector(y, 0);
+  if (check_ptr(rho, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+  sunrealtype* mx = N_VGetSubvectorArrayPointer_ManyVector(y, 1);
+  if (check_ptr(mx, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+  sunrealtype* my = N_VGetSubvectorArrayPointer_ManyVector(y, 2);
+  if (check_ptr(my, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+  sunrealtype* mz = N_VGetSubvectorArrayPointer_ManyVector(y, 3);
+  if (check_ptr(mz, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+  sunrealtype* et = N_VGetSubvectorArrayPointer_ManyVector(y, 4);
+  if (check_ptr(et, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+
+  sunrealtype* rhodot = N_VGetSubvectorArrayPointer_ManyVector(f, 0);
+  if (check_ptr(rhodot, "N_VGetSubvectorArrayPointer_ManyVector"))
+  {
+    return -1;
+  }
+  sunrealtype* mxdot = N_VGetSubvectorArrayPointer_ManyVector(f, 1);
+  if (check_ptr(mxdot, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+  sunrealtype* mydot = N_VGetSubvectorArrayPointer_ManyVector(f, 2);
+  if (check_ptr(mydot, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+  sunrealtype* mzdot = N_VGetSubvectorArrayPointer_ManyVector(f, 3);
+  if (check_ptr(mzdot, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+  sunrealtype* etdot = N_VGetSubvectorArrayPointer_ManyVector(f, 4);
+  if (check_ptr(etdot, "N_VGetSubvectorArrayPointer_ManyVector")) { return -1; }
+
+  // Set shortcut variables
+  const long int nx    = udata->nx;
```

Comment:
We generally do not put such requirements on our examples, since we do not recommend that users write their codes using `sunindextype` and `sunrealtype` (only that they compile SUNDIALS to match the data types they want in their code).  

I generally prefer to use `sunrealtype` in my own examples, if only because then we can use them as regression tests when building SUNDIALS with differing precisions.  I similarly only use `sunindextype` for arguments to SUNDIALS functions that could change type depending on how the library is built.